HeterogeneousCore/CUDACore/src/GPUCuda.cc

```diff
@@ -46,7 +46,7 @@ namespace heterogeneous {
     //   * being able to cheaply allocate+deallocate scratch memory allows to make the execution fully dynamic e.g. based on current load
     //   * would probably still need some buffer space/device to hold e.g. conditions data
     //     - for conditions, how to handle multiple lumis per job?
-    deviceId_ = id % cudaService->numberOfDevices();
+    deviceId_ = cudaService->devices()[id % cudaService->numberOfDevices()];
```

Comment:
@makortel should we replace this with a call to `cudacore::chooseCUDADevice(id)` ?

Reply:
Ok for me if you want to. This code will eventually be removed, but it might take a while.